src/nl/hannahsten/texifyidea/settings/conventions/TexifyConventionsConfigurable.kt

```diff
@@ -0,0 +1,233 @@
+    private lateinit var maxSectionSize: JBIntSpinner
+    private lateinit var labelConventionsTable: TableModelEditor<LabelConvention>
+
+    override fun createComponent(): JComponent {
+
+        schemesPanel = TexifyConventionsSchemesPanel(unsavedSettings)
+
+        /**
+         * Save the current scheme and load the new scheme whenever the combobox changes
+         */
+        schemesPanel.addListener(object : TexifyConventionsSchemesPanel.Listener {
+            override fun onCurrentSchemeWillChange(scheme: TexifyConventionsScheme) {
+                saveScheme(scheme)
+            }
+
+            override fun onCurrentSchemeHasChanged(scheme: TexifyConventionsScheme) {
+                loadScheme(scheme)
+            }
+        })
+
+        val prefixColumnInfo = object : TableModelEditor.EditableColumnInfo<LabelConvention, String>("Prefix") {
+            override fun valueOf(item: LabelConvention): String = item.prefix!!
+            override fun setValue(item: LabelConvention, value: String?) {
+                item.prefix = value ?: ""
+            }
+        }
+
+        val nameColumnInfo =
+            object : TableModelEditor.EditableColumnInfo<LabelConvention, LabelConvention>("Element") {
+                override fun valueOf(item: LabelConvention): LabelConvention = item
+                override fun isCellEditable(item: LabelConvention?): Boolean = false
+                override fun getColumnClass(): Class<*> = LabelConvention::class.java
+                override fun getRenderer(item: LabelConvention?): TableCellRenderer {
+                    return object : DefaultTableCellRenderer() {
+                        override fun getTableCellRendererComponent(
+                            table: JTable?,
+                            value: Any?,
+                            selected: Boolean,
+                            focus: Boolean,
+                            row: Int,
+                            column: Int
+                        ): Component {
+                            val convention = value as LabelConvention?
+                            super.getTableCellRendererComponent(table, convention?.name, selected, focus, row, column)
+                            //noinspection unchecked
+                            icon = if (value != null) {
+                                when (value.type!!) {
+                                    LabelConventionType.ENVIRONMENT -> TexifyIcons.DOT_ENVIRONMENT
+                                    LabelConventionType.COMMAND -> TexifyIcons.DOT_COMMAND
+                                }
+                            }
+                            else {
+                                null
+                            }
+
+                            return this
+                        }
+                    }
+                }
+            }
+
+        val enabledColumnInfo = object : TableModelEditor.EditableColumnInfo<LabelConvention, Boolean>("") {
```

Comment:
I think this should have a name as well, it's not clear what it is enabling (the inspection which check prefixes? whether it generates a prefix automatically or not?)
Maybe we should add a help text to clarify

Reply:
I added a column name (still looks a bit ugly, because the checkbox is centered) as well as a tooltip. The idea behind the tooltip was to control the `LatexMissingLabelInspection`. Such a control would also supersede the "minimum section level which triggers the missing label inspection" because with the checkbox a user could decide to have labels on a specific section level but not another one. If you prefer the current setting though we can also repurpose the checkbox to control the `LatexLabelConventionInspection` inspection instead or even have two checkboxes.

Reply:
Ah, well, that's why a column name was necessary I guess :D No I think the current purpose is fine.

Yes, of course the spacing should be fixed, and I think the column should be on the right instead of the left (so that the column with elements for which you change settings, is on the left)